src/py/py_multipatch.cpp

```diff
@@ -665,68 +665,103 @@ void GetBoundaryOrientation(
       const double cos_angle = abs(dot_p / std::sqrt(norm_s * norm_e));
       if (cos_angle > (1. - tolerance)) {
         int_mappings_ptr[i_pd] = j;
-        bool_orientations_ptr[i_pd] = (dot_p > 0);
+        bool_orientations_ptr[i_pd] = (dot_p > 0) ? 1 : 0;
         break;
       }
     }
   }
 }
 
-py::tuple GetBoundaryOrientations(const py::list& spline_list,
-                                  const py::array_t<int>& base_id,
-                                  const py::array_t<int>& base_face_id,
-                                  const py::array_t<int>& neighbor_id,
-                                  const py::array_t<int>& neighbor_face_id,
-                                  const double tolerance,
-                                  const int n_threads) {
-  // Basic Checks
-  // Check if all have same size
-  if (!((base_id.size() == base_face_id.size())
-        && (neighbor_id.size() == neighbor_face_id.size())
-        && (base_id.size() == neighbor_face_id.size()))) {
-    splinepy::utils::PrintAndThrowError(
-        "The ID arrays need to be of same size, please check for "
-        "consistencies.");
+py::array_t<int> PyMultipatch::GetInterfaceOrientations(const double tolerance,
+                                                        const int n_threads) {
+
+  // Check if already exists
+  if (orientations_.size() != 0) {
+    return orientations_;
   }
 
-  // Auxiliary data
-  const int* base_id_ptr = static_cast<int*>(base_id.request().ptr);
-  const int* base_face_id_ptr = static_cast<int*>(base_face_id.request().ptr);
-  const int* neighbor_id_ptr = static_cast<int*>(neighbor_id.request().ptr);
-  const int* neighbor_face_id_ptr =
-      static_cast<int*>(neighbor_face_id.request().ptr);
-  const auto cpp_spline_list = ToCoreSplineVector(spline_list);
-  const int n_connections = base_id.size();
+  int n_patches = patches_.size();
+
+  // get para dim of the first patch
+  int para_dim = ParaDim();
+
+  // Pointer request() for interfaces
+  const int n_faces_per_element = 2 * para_dim;
+  const int n_entries_per_line = 4 + 2 * para_dim;
+
+  // Make sure Interfaces are available
+  GetInterfaces(false);
+  const int* interfaces_ptr = static_cast<int*>(interfaces_.request().ptr);
+
+  // Loop over interfaces to compute number of interfaces (This is generally not
+  // time critical as comparisons are much more expensive)
+  int number_of_interfaces{};
+  for (int i_interface{}; i_interface < interfaces_.size(); i_interface++) {
+    if (interfaces_ptr[i_interface] >= 0) {
+      // interface found
+      if (interfaces_ptr[i_interface] > i_interface) {
+        number_of_interfaces++;
+      }
+    }
+  }
+
+  orientations_ = py::array_t<int>(number_of_interfaces * (4 + para_dim * 2));
```

Comment:
```suggestion
  orientations_ = py::array_t<int>({number_of_interfaces, (4 + para_dim * 2)});
```
we can avoid resize with this and is per default c-contiguous